spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,374 @@
+
+    it "should return true if a falsey value" do
+      values.each do |value|
+        Bool.from_yaml(value).should eq false
+      end
+    end
+
+    it "should raise if a falsey value is quoted with double quotes" do
+      values.each do |value|
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml %("#{value}")
+        end
+      end
+    end
+
+    it "should raise if falsey is quoted with single quotes" do
+      values.each do |value|
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml "'#{value}'"
+        end
+      end
+    end
+  end
+
+  it "should raise if not a bool value" do
+    expect_raises(YAML::ParseException) do
+      Bool.from_yaml "hello"
+    end
+  end
+end
+
+[Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64].each do |type|
+  describe "#{type.name}.from_yaml" do
+    it "should serialize and deserialize" do
+      type.from_yaml(type.new("1").to_yaml).should eq type.new("1")
+    end
+
+    it "should parse a number into an #{type.name}" do
+      type.from_yaml("1").should eq type.new("1")
+    end
+
+    it "should raise if quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml %("1")
+      end
+    end
+
+    it "should raise if quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml "'1'"
+      end
+    end
+
+    it "should raise when not a number" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml("hi")
+      end
+    end
+  end
+end
+
+[Float32, Float64].each do |type|
+  describe "#{type.name}.from_yaml" do
+    it "should serialize and deserialize" do
+      type.from_yaml(type.new("1.0").to_yaml).should eq type.new("1.0")
+    end
+
+    it "should parse a number into an #{type.name}" do
+      type.from_yaml("1").should eq type.new("1")
+    end
+
+    it "should parse a float into an #{type.name}" do
+      type.from_yaml("1.1").should eq type.new("1.1")
+    end
+
+    it "should raise if quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml %("1.1")
+      end
+    end
+
+    it "should raise if quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml "'1.1'"
+      end
+    end
+
+    it "should raise when not a float or number" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml("hi")
+      end
+    end
+  end
+end
+
+describe "String.from_yaml" do
+  it "should serialize and deserialize" do
+    String.from_yaml("hi".to_yaml).should eq "hi"
+  end
+
+  it "should parse a string" do
+    String.from_yaml("hello").should eq "hello"
+  end
+
+  it "should parse a single quoted string" do
+    String.from_yaml("'hello'").should eq "hello"
+  end
+
+  it "should parse a double quoted string" do
+    String.from_yaml(%("hello")).should eq "hello"
+  end
+
+  it "should parse a literal string" do
```

Comment:
There should be more detailed tests about the specifics of literal style and folded style values, at least for strings. As far as I can tell, there are none in the stdlib specs as of now. Would be nice to add them here.

Reply:
How would you like me to add specifics?

Reply:
For example
```crystal
String.from_yaml("|\n  hello\n  world").should eq "hello\nworld"
# and
String.from_yaml(">\n  hello\  world").should eq "hello world"
```